### DISCUSSION TITLE
Realtime: Channel usage in swift

### DISCUSSION THREAD

**Discussion by zigzagg16** [supabase/supabase#12481] [Questions]
Hi there,

I'm trying to setup realtime on a swift iOS app.
Unfortunately there's very few documentation, and I can't get it working.

Here's my code :

```
var channel: Realtime.Channel {
	let topic = ChannelTopic(rawValue: "channel")!
	let channel = client.realtime.channel(topic)
	return channel
}

public func sendRealtime() {
	client.realtime.connect()
	channel.subscribe()
	channel.on(.all) { message in
		print(message)
	}
	channel.push(.all, payload: ["hey":"test"])
}
```

I get a fatal error : `Fatal error: Tried to push all to schema("channel") before joining. Use channel.join() before pushing events
(lldb) `

Any idea what's missing here ?

**Comment by mrardyan** [comment]
I got a problem with supabase-swift real-time channel connection. I couldn't catch any events from my table. I just stumbled upon your issues and am curious about the way you successfully subscribed to the channel. If you don't mind could you drop your code here?

**Comment by luisazmouz** [comment]
Did you figure it out?
